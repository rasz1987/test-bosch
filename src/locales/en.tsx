import Locales from './Interfaces';

const en: Locales = {
  translation: {
    title: 'My title',
    notFound: 'Page not found'
  }
};

export default en;