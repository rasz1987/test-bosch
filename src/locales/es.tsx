import Locales from './Interfaces';

const es: Locales = {
  translation: {
    title: 'Mi titulo',
    notFound: 'PÃ¡gina no encontrada'
  }
};

export default es;