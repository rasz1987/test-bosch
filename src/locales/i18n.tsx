import { initReactI18next } from "react-i18next";
import i18n from "i18next";
import EN from './en';
import ES from './es';
import PT from './pt';

const fallbackLng = ['pt'];
const resources = {
  en: { ...EN },
  es: { ...ES },
  pt: { ...PT }
};

i18n
  .use(initReactI18next)
  .init({
    resources,
    fallbackLng,
    lng: 'pt',
    interpolation: {
      escapeValue: false
    }
  });

export default i18n;
