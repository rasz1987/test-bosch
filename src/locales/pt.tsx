import Locales from './Interfaces';

const pt: Locales = {
  translation: {
    title: 'O meu titulo',
    notFound: 'Página não encontrada',
    name: 'Nome',
    birthday: 'Aniversário',
    hasContract: 'Contratado',
    actions: 'Acções',
    yes: 'Sim',
    not: 'Não'
  }
};

export default pt;