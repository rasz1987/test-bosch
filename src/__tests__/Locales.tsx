import en from '../locales/en';
import es from '../locales/es';
import pt from '../locales/pt';
import i18n from '../locales/i18n';

test('Should return locales without crash', () => {
  expect(en).toMatchSnapshot();
  expect(es).toMatchSnapshot();
  expect(pt).toMatchSnapshot();
  expect(i18n).toMatchSnapshot();
});