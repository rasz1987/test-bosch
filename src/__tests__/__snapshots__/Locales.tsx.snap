// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return locales without crash 1`] = `
Object {
  "translation": Object {
    "actions": "Actions",
    "back": "Back",
    "birthday": "Birthday",
    "email": "Email",
    "hasContract": "Has contract",
    "lastname": "Last name",
    "name": "First name",
    "not": "Not",
    "notFound": "Page not found",
    "yes": "Yes",
  },
}
`;

exports[`Should return locales without crash 2`] = `
Object {
  "translation": Object {
    "actions": "Acciones",
    "back": "Regresar",
    "birthday": "Cumpleaños",
    "email": "Correo electrónico",
    "hasContract": "Contratado",
    "lastname": "Apellido",
    "name": "Primer nombre",
    "not": "No",
    "notFound": "Página no encontrada",
    "yes": "Sí",
  },
}
`;

exports[`Should return locales without crash 3`] = `
Object {
  "translation": Object {
    "actions": "Acções",
    "back": "Voltar",
    "birthday": "Aniversário",
    "email": "Correio eletronico",
    "hasContract": "Contratado",
    "lastname": "Sobrenome",
    "name": "Nome",
    "not": "Não",
    "notFound": "Página não encontrada",
    "yes": "Sim",
  },
}
`;

exports[`Should return locales without crash 4`] = `
Object {
  "language": "pt",
  "languages": Array [
    "pt",
  ],
  "options": Object {
    "appendNamespaceToCIMode": false,
    "appendNamespaceToMissingKey": false,
    "contextSeparator": "_",
    "debug": false,
    "defaultNS": Array [
      "translation",
    ],
    "fallbackLng": Array [
      "pt",
    ],
    "fallbackNS": false,
    "ignoreJSONStructure": true,
    "initImmediate": true,
    "interpolation": Object {
      "escapeValue": false,
      "format": [Function],
      "formatSeparator": ",",
      "maxReplaces": 1000,
      "nestingOptionsSeparator": ",",
      "nestingPrefix": "$t(",
      "nestingSuffix": ")",
      "prefix": "{{",
      "skipOnVariables": true,
      "suffix": "}}",
      "unescapePrefix": "-",
    },
    "joinArrays": false,
    "keySeparator": ".",
    "lng": "pt",
    "load": "all",
    "missingInterpolationHandler": false,
    "missingKeyHandler": false,
    "nonExplicitSupportedLngs": false,
    "ns": Array [
      "translation",
    ],
    "nsSeparator": ":",
    "overloadTranslationOptionHandler": [Function],
    "parseMissingKeyHandler": false,
    "partialBundledLanguages": false,
    "pluralSeparator": "_",
    "postProcess": false,
    "postProcessPassResolved": false,
    "preload": false,
    "resources": Object {
      "en": Object {
        "translation": Object {
          "actions": "Actions",
          "back": "Back",
          "birthday": "Birthday",
          "email": "Email",
          "hasContract": "Has contract",
          "lastname": "Last name",
          "name": "First name",
          "not": "Not",
          "notFound": "Page not found",
          "yes": "Yes",
        },
      },
      "es": Object {
        "translation": Object {
          "actions": "Acciones",
          "back": "Regresar",
          "birthday": "Cumpleaños",
          "email": "Correo electrónico",
          "hasContract": "Contratado",
          "lastname": "Apellido",
          "name": "Primer nombre",
          "not": "No",
          "notFound": "Página no encontrada",
          "yes": "Sí",
        },
      },
      "pt": Object {
        "translation": Object {
          "actions": "Acções",
          "back": "Voltar",
          "birthday": "Aniversário",
          "email": "Correio eletronico",
          "hasContract": "Contratado",
          "lastname": "Sobrenome",
          "name": "Nome",
          "not": "Não",
          "notFound": "Página não encontrada",
          "yes": "Sim",
        },
      },
    },
    "returnEmptyString": true,
    "returnNull": true,
    "returnObjects": false,
    "returnedObjectHandler": false,
    "saveMissing": false,
    "saveMissingPlurals": true,
    "saveMissingTo": "fallback",
    "simplifyPluralSuffix": true,
    "supportedLngs": false,
    "updateMissing": false,
  },
  "resolvedLanguage": "pt",
  "store": Object {
    "en": Object {
      "translation": Object {
        "actions": "Actions",
        "back": "Back",
        "birthday": "Birthday",
        "email": "Email",
        "hasContract": "Has contract",
        "lastname": "Last name",
        "name": "First name",
        "not": "Not",
        "notFound": "Page not found",
        "yes": "Yes",
      },
    },
    "es": Object {
      "translation": Object {
        "actions": "Acciones",
        "back": "Regresar",
        "birthday": "Cumpleaños",
        "email": "Correo electrónico",
        "hasContract": "Contratado",
        "lastname": "Apellido",
        "name": "Primer nombre",
        "not": "No",
        "notFound": "Página no encontrada",
        "yes": "Sí",
      },
    },
    "pt": Object {
      "translation": Object {
        "actions": "Acções",
        "back": "Voltar",
        "birthday": "Aniversário",
        "email": "Correio eletronico",
        "hasContract": "Contratado",
        "lastname": "Sobrenome",
        "name": "Nome",
        "not": "Não",
        "notFound": "Página não encontrada",
        "yes": "Sim",
      },
    },
  },
}
`;
